packages/react-browser-pathname/src/index.tsx

```diff
@@ -1,24 +1,42 @@
 import React, { useContext, useCallback, useEffect, useState } from 'react';
 import type { ReactNode } from 'react';
 
-import { PathnameContext, UpdateContext } from './context';
+import { PathnameContext, UpdateContext, SET_ACTION } from './context';
 
 const getPathname = () => `${window.location.pathname}`;
 
-const BrowserPathname = ({ children }: { children: ReactNode }) => {
-  const [location, setLocation] = useState(getPathname());
+const BrowserPathname = ({
+  children,
+  initialPath,
+}: {
+  children: ReactNode;
+  initialPath: string;
+}) => {
+  const [location, setLocation] = useState(initialPath);
 
-  const setCurrentBrowserPathname = useCallback(() => {
-    setLocation(getPathname());
-  }, []);
+  const setCurrentBrowserPathname = useCallback(
+    ({ data, title, url, type }: SET_ACTION) => {
+      switch (type) {
+        case 'REPLACE':
+          window.history.replaceState(data, title, url);
+          break;
+        case 'PUSH':
+          window.history.pushState(data, title, url);
```

Comment:
title doesn't do anything. https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

Reply:
lol this api is so dumb, why is the unused thing the second arg